addons/mail/static/src/messaging/component/message/message_tests.js

```diff
@@ -217,6 +219,131 @@ QUnit.test('moderation: moderated channel with pending moderation message (moder
+        '.o_Message_notificationIcon',
+        "should display the notification icon"
+    );
+    assert.hasClass(
+        document.querySelector('.o_Message_notificationIcon'),
+        'fa-envelope-o',
+        "icon should represent email success"
+    );
+
+    await afterNextRender(() => {
+        document.querySelector('.o_Message_notificationIconContainer').click();
+    });
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationPopover',
+        "notification popover should be open"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationPopover_icon',
+        "popover should have one icon"
+    );
+    assert.hasClass(
+        document.querySelector('.o_NotificationPopover_icon'),
+        'fa-check',
+        "popover should have the sent icon"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_NotificationPopover_partnerName',
+        "popover should have the partner name"
+    );
+    assert.equal(
+        document.querySelector('.o_NotificationPopover_partnerName').textContent.trim(),
+        "Someone",
+        "partner name should be correct"
+    );
+});
+
+QUnit.test('Notification Error', async function (assert) {
+    assert.expect(8);
+
+    await this.start({
+        intercepts: {
+            do_action(ev) {
+                assert.step('do_action');
+                assert.strictEqual(
+                    ev.data.action,
+                    'mail.mail_resend_message_action',
+                    "action should be the one to cancel email"
+                );
+                assert.strictEqual(
+                    ev.data.options.additional_context.mail_message_to_resend,
+                    10,
+                    "action should have correct message id"
+                );
+            },
+        },
+    });
+    const message = this.env.entities.Message.create({
+        id: 10,
+        message_type: 'email',
+        notifications: [['insert', {
+            id: 11,
+            notification_status: 'exception',
+            notification_type: 'email',
+        }]],
+    });
+    await this.createMessageComponent(message);
+
+    assert.containsOnce(
+        document.body,
+        '.o_Message',
+        "should display a message component"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_notificationIconContainer',
+        "should display the notification icon container"
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_Message_notificationIcon',
+        "should display the notification icon"
+    );
+    assert.hasClass(
+        document.querySelector('.o_Message_notificationIcon'),
+        'fa-envelope',
+        "icon should represent email error"
+    );
+
+    await afterNextRender(() => {
+        document.querySelector('.o_Message_notificationIconContainer').click();
+    });
+    assert.verifySteps(['do_action'], "'do_action' should have been called");
```

Comment:
I think I would clarify intent of do action in comment of assertion, something like "should have made an action to resend message", just to better understand the importance of `do_action` in this test, and should really help in future to adapt/remove them (like action manager is incorporated in mock environment, or we just stop using do_action and use something else, etc...).

I know assertions in intercepted do_action give these clues, but if test fails due to do_action not being intercepted for some reasons, it requires some effort to find appropriate information to understand why do_action is needed. Tests has some assertions inside intercepted `do_action` that give some clues to purpose of this `do_action`, but it still doesn't give a big picture of intent of this flow IMO, hence the suggestion to clarify in comment of this assertion.

Reply:
Yes it's better, I think I was just lazy to write better message since as you said assert inside action were already descriptive.